G-Research.OpenTelemetry.Processor.Partial/ActivitySpec.cs

```diff
@@ -0,0 +1,90 @@
+using System.Diagnostics;
+using System.Globalization;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+
+namespace GR.OpenTelemetry.Processor.Partial
+{
+    public class ActivitySpec
+    {
+        public string Name { get; set; }
+        public Context Context { get; set; }
+        public string ParentId { get; set; }
+        public string StartTime { get; set; }
+        public string EndTime { get; set; }
+        public string StatusCode { get; set; }
+        public string StatusMessage { get; set; }
+        public Dictionary<string, object> Attributes { get; set; }
+        public List<Event> Events { get; set; }
+
+        public enum Signal
+        {
+            Heartbeat,
+            Stop
+        }
+
+        public ActivitySpec(Activity activity, Signal signal)
+        {
+            Name = activity.DisplayName;
+            Context = new Context(activity.Context);
+            ParentId = activity.ParentSpanId == default
+                ? string.Empty
+                : activity.ParentSpanId.ToString();
+            StartTime = FormatTimestamp(activity.StartTimeUtc);
+            EndTime = signal == Signal.Heartbeat
+                ? string.Empty
+                : FormatTimestamp(activity.StartTimeUtc.Add(activity.Duration));
+            StatusCode = activity.Status.ToString();
+            StatusMessage = activity.StatusDescription ?? string.Empty;
+            Attributes = activity.TagObjects
+                .ToDictionary(tag => tag.Key, tag => tag.Value)!;
+            Events = activity.Events
+                .Select(e => new Event
+                {
+                    Name = e.Name,
+                    Timestamp = FormatTimestamp(e.Timestamp.DateTime),
+                    Attributes = e.Tags.ToDictionary(tag => tag.Key, tag => tag.Value)!
+                }).ToList();
+        }
+
+        public static string FormatTimestamp(DateTime dateTime)
+        {
+            return dateTime.ToString("yyyy-MM-dd HH:mm:ss.ffffff",
+                CultureInfo.InvariantCulture) + " +0000 UTC";
+        }
```

Comment:
Why the manual formatting? Is the [built-in formatting](https://learn.microsoft.com/en-us/dotnet/standard/datetime/system-text-json-support#support-for-formatting) not compatible with the spec?

Reply:
this format is needed `yyyy-MM-dd HH:mm:ss.ffffff +0000 UTC` and there seems to be no built-in mechanism for this exact formatting, that's why ` +0000 UTC` is added manually.

Reply:
this is no longer valid